pkg/pull/pull.go

```diff
@@ -391,21 +391,24 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 	}
 
 	helmMidstreams := []midstream.Midstream{}
-	for _, base := range helmBases {
+	for _, helmBase := range helmBases {
 		writeMidstreamOptions := commonWriteMidstreamOptions
-		writeMidstreamOptions.MidstreamDir = filepath.Join(base.GetOverlaysDir(writeBaseOptions), "midstream", base.Path)
-		writeMidstreamOptions.BaseDir = filepath.Join(u.GetBaseDir(writeUpstreamOptions), base.Path)
-
-		helmBase := base
+		writeMidstreamOptions.MidstreamDir = filepath.Join(helmBase.GetOverlaysDir(writeBaseOptions), "midstream", helmBase.Path)
+		writeMidstreamOptions.BaseDir = filepath.Join(u.GetBaseDir(writeUpstreamOptions), helmBase.Path)
 
 		helmMidstream, err := writeMidstream(writeMidstreamOptions, pullOptions, u, &helmBase, fetchOptions.License, identityConfig, u.GetUpstreamDir(writeUpstreamOptions), log)
 		if err != nil {
-			return "", errors.Wrapf(err, "failed to write helm midstream %s", base.Path)
+			return "", errors.Wrapf(err, "failed to write helm midstream %s", helmBase.Path)
 		}
 
 		helmMidstreams = append(helmMidstreams, *helmMidstream)
 	}
 
+	err = removeUnusedHelmOverlays(writeMidstreamOptions.MidstreamDir, writeMidstreamOptions.BaseDir)
```

Comment:
wont this just be set to the last in the loop above?

Reply:
nah, that variable is in the loop scope.